source/adapters/level_zero/event.cpp

```diff
@@ -872,9 +872,11 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventRetain(
 UR_APIEXPORT ur_result_t UR_APICALL urEventRelease(
     ur_event_handle_t Event ///< [in] handle of the event object
 ) {
-  Event->RefCountExternal--;
-  UR_CALL(urEventReleaseInternal(Event));
-
+  if (Event->CounterBasedEventsEnabled && --Event->RefCountExternal == 0) {
```

Comment:
If Event->CounterBasedEventsEnabled == false, won't the decrement not take place?  The second check call won't take place if the first is false.

We need to decrement the event refcnt both if counter based events is enabled and when it is not.